### PR TITLE
JSXGenerator refactor ⚙

### PR DESCRIPTION
**Pull request by anna-wro**
no need to pass propsWithDefaults to JSXGenerator

### CODE REVIEW

File: src/docs/components/DocsActiveBlock.jsx [hunk 1]
```diff
@@ -61,22 +61,35 @@ class DocsActiveBlock extends Component {
...
       });
 
-    return Object.keys(component.props)
-      .filter(key => unchangedProps.indexOf(key) === -1);
+    return Object.keys(props)
+      .filter(key => unchangedProps.indexOf(key) > -1);
+  }
+
+  removePropsWithDefaultValues(props) {
+    const fewerProps = Object.assign(props);
```
Comment (matikrk) [inline src/docs/components/DocsActiveBlock.jsx]: `Object.assign({},props); `
to make copy

File: src/docs/components/DocsActiveBlock.jsx [hunk 2]
```diff
@@ -61,22 +61,35 @@ class DocsActiveBlock extends Component {
...
 
     if (!Array.isArray(settings)) {
       return [];
     }
 
-    const unchangedProps = Object.keys(component.props)
+    const unchangedProps = Object.keys(props)
       .filter(key => {
-        const propSettings = settings.find(setting => setting.name === key);
+        const propSettings = settings.find(setting => setting.name === key.value);
```
Comment (matikrk) [inline src/docs/components/DocsActiveBlock.jsx]: why we have this change?

Comment (anna-wro) [inline src/docs/components/DocsActiveBlock.jsx]: I changed it back to `setting.name === key`

File: src/docs/components/DocsActiveBlock.jsx (hunk 1 above)
Comment (anna-wro) [inline src/docs/components/DocsActiveBlock.jsx]: Fixed, thanks!

**Pull request merged**